### PR TITLE
Fix argument unused warning when building compiler-rt. NFC

### PR DESCRIPTION
**Pull request by sbc100**
Also ensure that warning are errors even when building asm files.

### CODE REVIEW

File: tools/system_libs.py [hunk 1]
```diff
@@ -49,14 +49,16 @@ def glob_in_path(path, glob_pattern, excludes=()):
   return sorted(f for f in files if os.path.basename(f) not in excludes)
 
 
-def get_base_cflags(force_object_files=False):
+def get_base_cflags(force_object_files=False, asm=False):
```
Comment (kripken) [inline tools/system_libs.py:52]: What does "asm" mean in this context? I can't make sense of the name "asm" to the usage below which adds a dynamic linking flag.

Comment (sbc100) [reply tools/system_libs.py:52]: It means "assembly file"  .. i.e. `-D` and `-I` flags should not be added

Comment (sbc100) [reply tools/system_libs.py]: I renamed it to "preprocess" to be more clear.

**Review by kripken — APPROVED** [review]

File: tools/system_libs.py [hunk 2]
```diff
@@ -842,6 +844,7 @@ class MuslInternalLibrary(Library):
     '-std=c99',
     '-D_XOPEN_SOURCE=700',
     '-Wno-unused-result',  # system call results are often ignored in musl, and in wasi that warns
+    '-Wno-missing-braces',  # system call results are often ignored in musl, and in wasi that warns
```
Comment (kripken) [inline tools/system_libs.py:847]: comment looks copied from the last line

Comment (sbc100) [reply tools/system_libs.py]: Oops, reverted

**Pull request merged**